src/tests/sync/src/test_utils.rs

```diff
@@ -0,0 +1,155 @@
+    std::env::set_var("PLAYGROUND_PROFILE", "playground-test");
+    std::env::set_var("OBJECT_STORE_URL", "memory-shared");
+    std::env::set_var("OBJECT_STORE_BUCKET", "hummock_001");
+}
+
+pub async fn start_cluster() -> (JoinHandle<()>, std::sync::mpsc::Sender<()>) {
+    wipe_object_store().await;
+    sync::activate_sync_point(
+        "CLUSTER_READY",
+        vec![sync::Action::EmitSignal("SIG_CLUSTER_READY".to_owned())],
+        1,
+    );
+
+    let (tx, rx) = std::sync::mpsc::channel();
+    let join_handle = std::thread::spawn(move || {
+        let runtime = tokio::runtime::Builder::new_multi_thread()
+            .worker_threads(1)
+            .enable_all()
+            .build()
+            .unwrap();
+        runtime.block_on(async {
+            tokio::spawn(async { playground().await });
+            rx.recv().unwrap();
+        });
+    });
+    // It will find "SIG_CLUSTER_READY" even when it is reached after "SIG_CLUSTER_READY" has been
+    // emitted.
+    wait_now("SIG_CLUSTER_READY".to_owned(), Duration::from_secs(30))
+        .await
+        .unwrap();
+    (join_handle, tx)
+}
+
+pub async fn stop_cluster(join_handle: JoinHandle<()>, shutdown_tx: std::sync::mpsc::Sender<()>) {
+    shutdown_tx.send(()).unwrap();
+    join_handle.join().unwrap();
+    wipe_object_store().await;
+}
+
+pub fn run_slt() -> Output {
+    Command::new("./risedev")
+        .args([
+            "slt",
+            "-d",
+            "dev",
+            "-p",
+            "4566",
+            "src/tests/sync/slt/tpch_snapshot_no_drop.slt",
+        ])
+        .spawn()
+        .unwrap()
+        .wait_with_output()
+        .unwrap()
+}
+
+pub async fn get_object_store_client() -> ObjectStoreImpl {
+    let url = std::env::var("OBJECT_STORE_URL").unwrap();
+    parse_remote_object_store(&url, Arc::new(ObjectStoreMetrics::unused())).await
+}
+
+pub fn get_object_store_bucket() -> String {
+    std::env::var("OBJECT_STORE_BUCKET").unwrap()
+}
+
+async fn wipe_object_store() {
+    // TODO: wipe content of object store
+}
+
+pub async fn get_meta_client() -> MetaClient {
+    let meta_addr = std::env::var("RW_META_ADDR").unwrap();
+    let mut client = MetaClient::new(&meta_addr).await.unwrap();
+    let worker_id = client
+        .register(WorkerType::Generic, &"127.0.0.1:2333".parse().unwrap(), 0)
+        .await
+        .unwrap();
+    client.set_worker_id(worker_id);
+    client
+}
+
+pub async fn wait_now(signal: Signal, timeout: Duration) -> Result<(), sync::Error> {
+    sync::activate_sync_point(
+        "NOW",
+        vec![sync::Action::WaitForSignal(WaitForSignal {
+            signal,
+            clear_signal: true,
+            timeout,
+        })],
+        1,
+    );
+    sync::on_sync_point("NOW").await
+}
+
+pub async fn emit_now(signal: Signal) {
+    sync::activate_sync_point("NOW", vec![sync::Action::EmitSignal(signal)], 1);
+    sync::on_sync_point("NOW").await.unwrap();
+}
+
+#[tokio::test]
+#[serial]
```

Comment:
Perhaps to avoid serial requirement, we can introduce the concept of a "session", somehow, rather than rely on global var in the form of lazy staic.

Reply:
One idea is to use a task-local variable, and make sure that every spawn that contains a sync point will try to initialize its task-local session variable from its parent.

Reply:
since tokio uses a runtime per test (see: https://github.com/tokio-rs/tokio/issues/2374) it may be sufficient to define a runtime-local variable

Reply:
One possible hack is to use the memory address of the tokio Handle (see: https://docs.rs/tokio/latest/src/tokio/runtime/handle.rs.html#17-19) which is unique to each runtime as the session ID.

Reply:
I've created a forum question here to verify this hypothesis: https://users.rust-lang.org/t/handleinner-memory-address-as-tokio-runtime-unique-identifier/80314

Reply:
On second thouhgt, this may not work as the methods required are not public. However, perhaps, since all our services connect to meta, we could simply use meta address+port as session identifier?

Reply:
Actually, since #[tokio::test] is single-threaded, simply initializing a thread-local variable seems to be sufficient. I'm quite surprised that the above test runs on a single thread...!